lib/matplotlib/axis.py

```diff
@@ -1477,9 +1479,14 @@ def have_units(self):
         return self.converter is not None or self.units is not None
 
     def convert_units(self, x):
-        # If x is already a number, doesn't need converting
+        # If x is already a number, doesn't need converting, but
+        # some iterables need to be massaged a bit...
         if munits.ConversionInterface.is_numlike(x):
-            return x
+            if isinstance(x, list) or isinstance(x, numbers.Number):
+                return x
+            if issubclass(type(x), np.ma.MaskedArray):
+                return np.ma.asarray(x)
+            return np.asarray(x)
```

Comment:
it seems worse, because calling `np.asarray` can drop information in some cases, making it unavailable in the call to `convert` below. Unless I'm missing something...